src/loaders/workspace_uploader/workspace_uploader.py

```diff
@@ -318,129 +444,147 @@ def _post_process(
 
-    # Put the assembly files in the input_queue
-    counter = 1
-    for assembly_name, host_assembly_dir in wait_to_upload_assemblies.items():
+    uploaded_count = 0
+    uploaded_fail = False
+    for assembly_tuples in _gen(wait_to_upload_assemblies, batch_size):
+        batch_upas = tuple()
+        batch_uploaded_tuples = []
 
-        container_internal_assembly_path = os.path.join(
-            JOB_DIR_IN_ASSEMBLYUTIL_CONTAINER, assembly_name
-        )
-        conf.input_queue.put(
-            (
+        try:
+            batch_upas = _upload_assemblies_to_workspace(
+                asu, workspace_id, load_id, assembly_tuples
+            )
+            batch_uploaded_tuples = assembly_tuples
+        except Exception as e:
+            print(e)
+            uploaded_fail = True
+
+            try:
+                # figure out uploads that succeeded
+                name2tuple = {
+                    assembly_tuple.assembly_name: assembly_tuple
+                    for assembly_tuple in assembly_tuples
+                }
+                (
+                    uploaded_obj_names,
+                    uploaded_obj_upas,
+                ) = _query_workspace_with_load_id_mass(
+                    ws, workspace_id, load_id, list(name2tuple.keys())
+                )
+
+                batch_upas = tuple(uploaded_obj_upas)
+                batch_uploaded_tuples = [
+                    name2tuple[name] for name in uploaded_obj_names
+                ]
+
+            except Exception as e:
+                print(
+                    f"WARNING: There are inconsistencies between "
+                    f"the workspace and the yaml files as the result of {e}"
+                )
+
+        # post process on sucessful uploads
+        for assembly_tuple, upa in zip(batch_uploaded_tuples, batch_upas):
+            _post_process(
                 upload_env_key,
                 workspace_id,
-                container_internal_assembly_path,
-                host_assembly_dir,
-                assembly_name,
+                load_id,
+                assembly_tuple,
                 upload_dir,
-                counter,
-                assembly_files_len,
+                output_dir,
+                upa,
             )
-        )
 
-        if counter % 5000 == 0:
-            print(f"Jobs added to the queue: {counter}/{assembly_files_len}, "
-                  f"Percentage: {counter / assembly_files_len * 100:.2f}%, "
-                  f"Time: {datetime.now()}")
+        uploaded_count += len(batch_uploaded_tuples)
+        if uploaded_count % 100 == 0:
+            print(
+                f"Assemblies uploaded: {uploaded_count}/{assembly_files_len}, "
+                f"Percentage: {uploaded_count / assembly_files_len * 100:.2f}%, "
+                f"Time: {datetime.now()}"
+            )
 
-        counter += 1
+        if uploaded_fail:
+            return uploaded_count
```

Comment:
At some point (not now) this code could probably just keep going on a failure as long as the recovery succeeds

Reply:
I want the client still be able to see the upload status (like time, counts, average rate) at the end for the current iteration. That's why.

Reply:
Nothing preventing printing that out on a recovery

Reply:
👍

Reply:
Nothing to do here so :+1: